## 88% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 1
Expected to fail tests: 0
Passed tests: 7
```
```
IDs of failed tests: 703133 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=send-collectible-route-id&PR_ID=21744&APK_NAME=21744.apk&TR_CASE_IDS=703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740659">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:240: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aa8c6ec4eb7414f8df7338e92d7d012?auth=1db01e77dacb6ee9dfb877d412c85add#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (7)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740661">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740660">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740662">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740663">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740658">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0ae58dd8fcb74cf1a68da354a2467a41?auth=563ec861e07bc6d1c087861b244b865a#171">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9635f4ae42394f03bd475421c6ab9ccc?auth=9e2266f13fb745c9ec7a25138262d277#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740657">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7fe11b0ae15a405b86a36a4b6477581b?auth=88ddafa9a641bb12cd868ced93cfb7f6#104">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4779668de8284b368aaad3ee91f8b755?auth=cfa4ab5e92d867dfc6b12980f8eae029#95">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3740656">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5aa8c6ec4eb7414f8df7338e92d7d012?auth=1db01e77dacb6ee9dfb877d412c85add#47">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/send/events.cljs
+++ b/src/status_im/contexts/wallet/send/events.cljs
@@ -337,6 +337,13 @@
               :flow-id        :wallet-send-flow}]]]})))
 
 (rf/reg-event-fx
+ :wallet/build-transaction-for-collectible-route
+ (fn [{:keys [db]}]
+   (let [last-request-uuid (get-in db [:wallet :ui :send :last-request-uuid])]
+     {:db (update-in db [:wallet :ui :send] dissoc :transaction-for-signing)
+      :fx [[:dispatch [:wallet/build-transactions-from-route {:request-uuid last-request-uuid}]]]})))
+
+(rf/reg-event-fx
  :wallet/set-token-amount-to-bridge
  (fn [{:keys [db]} [{:keys [amount stack-id start-flow?]}]]
    (let [last-request-uuid (get-in db [:wallet :ui :send :last-request-uuid])]

--- a/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
+++ b/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
@@ -252,7 +252,7 @@
         ;; In token send flow we already have transaction built when
         ;; we reach confirmation screen. But in send collectible flow
         ;; routes request happens at the same time with navigation to
-        ;; confirmation screen. So we need to build the transaction ass soon
+        ;; confirmation screen. So we need to build the transaction as soon
         ;; as route is available.
         (rn/use-effect
          (fn []

--- a/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
+++ b/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
@@ -256,7 +256,9 @@
         ;; as route is available.
         (rn/use-effect
          (fn []
-           (when (and (= transaction-type :tx/collectible-erc-1155) first-route)
+           (when (and (or (= transaction-type :tx/collectible-erc-1155)
+                          (= transaction-type :tx/collectible-erc-721))
+                      first-route)
              (rf/dispatch [:wallet/build-transaction-for-collectible-route])))
          [first-route])
         [rn/view {:style {:flex 1}}

--- a/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
+++ b/src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs
@@ -249,6 +249,16 @@
             user-props                {:full-name to-address
                                        :address   (utils/get-shortened-address
                                                    to-address)}]
+        ;; In token send flow we already have transaction built when
+        ;; we reach confirmation screen. But in send collectible flow
+        ;; routes request happens at the same time with navigation to
+        ;; confirmation screen. So we need to build the transaction ass soon
+        ;; as route is available.
+        (rn/use-effect
+         (fn []
+           (when (and (= transaction-type :tx/collectible-erc-1155) first-route)
+             (rf/dispatch [:wallet/build-transaction-for-collectible-route])))
+         [first-route])
         [rn/view {:style {:flex 1}}
          [floating-button-page/view
           {:footer-container-padding 0

